src/models/grouping.interface.ts

```diff
@@ -3,7 +3,7 @@ import { GroupingComparerItem } from "./groupingComparerItem.interface";
 import { GroupingFormatterItem } from "./groupingFormatterItem.interface";
 import { SortDirectionNumber } from "./sortDirectionNumber.enum";
 
-export type GroupingGetterFunction<T = any> = (value: T) => string;
+export type GroupingGetterFunction<T = any> = (value: T) => any;
```

Comment:
why are you converting this to `any`?

Reply:
I will open two separate PRs  (one for GroupItemMetaDataProvider and other for GroupingGetterFunction. I accidentally created MR from our master branch

About `any`; not sure if it is the right approach; GroupingGetterFunction is free to return anything I guess, only equality of the returned values is taken into account in the underlying slickgrid logic. I guess it is valid that grouping getter returns `number` for some rows `string` for others? Not sure how to type that differently than `any`? Maybe `unknown` would be better to force explicit typing when specific return type of grouper might be needed?

Reply:
I again disagree, it should always be a string, we're always looking to return a column name (name). This is not about what is being grouped, it's about the column name to find via the `getter` property of the group. Unless I missed something, if so prove me wrong because I won't accept this being changed from `string`